DOCS-3179 iOS RUM Data Collected Edits (#777)
DataDog/dd-sdk-ios: docs/rum_collection/data_collected.md

    tag: "Documentation"
    text: "Datadog Real User Monitoring"
---

## Overview

The RUM iOS SDK generates events that have associated metrics and attributes. Metrics are quantifiable values that can be used for measurements related to the event. Attributes are non-quantifiable values used to slice metrics data (group by) in analytics. 

Every RUM event has all of the [default attributes](#default-attributes), for example, the device type (`device.type`) and user information such as their name (`usr.name`) and their country (`geo.country`).

There are additional [metrics and attributes that are specific to a given event type](#event-specific-metrics-and-attributes). For example, the metric `view.time_spent` is associated with "view" events and the attribute `resource.method` is associated with "resource" events.

| Event Type | Retention | Description                         |
|------------|-----------|-------------------------------------|

> | Session    | 30 days   | A session represents a real user journey on your mobile application. It begins when the user launches the application, and the session remains live as long as the user stays active. During the user journey, all RUM events generated as part of the session will share the same `session.id` attribute. |

Reviewer: We might want to add a note that the session is reset after 15 min of inactivity, and can be reset when the app is in the background if it gets killed by the OS

Reviewer: The 15 minutes inactivity yes, but the "app being killed" only applies to mobile

> | View       | 30 days   | A view represents a unique screen (or screen segment) on your mobile application. Individual `UIViewControllers` are classified as distinct views. While a user stays on a view, RUM event attributes (Errors, Resources, Actions) get attached to the view with a unique `view.id`.                           |

Reviewer: It might be interesting to include the exact lifecycle events we use to define when a view starts and ends here. (cc @ncreated @maxep )

Reviewer 2: Good idea 👍. The way it works is that View start and stop are notified by `viewDidAppear(animated:)` and `viewDidDisappear(animated:)` callbacks (on `UIViewController` class).

Author: ```suggestion
| View       | 30 days   | A view represents a unique screen (or screen segment) on your mobile application. A view starts and stops when the `viewDidAppear(animated:)` and `viewDidDisappear(animated:)` callbacks on the UIViewController class are notified. Individual `UIViewControllers` are classified as distinct views. While a user stays on a view, RUM event attributes (Errors, Resources, Actions) get attached to the view with a unique `view.id`.                           |
```

> | Resource   | 15 days   | A resource represents network requests to first-party hosts, APIs, 3rd party providers, and libraries in your mobile application. All requests generated during a user session are attached to the view with a unique `resource.id`.                                                                       |

Reviewer: What does the `and libraries` refer to? I guess it might be related to how libraries work in JS/Browser, but that doesn't translate well to iOS apps

Author: ```suggestion
| Resource   | 15 days   | A resource represents network requests to first-party hosts, APIs, and third-party providers in your mobile application. All requests generated during a user session are attached to the view with a unique `resource.id`.                                                                       |
```

> | Error      | 30 days   | An error represents an exception emitted by the mobile application attached to the view it is generated in.                                                                                                                                                                                        |

Reviewer: ```suggestion
| Error      | 30 days   | An error represents an exception emitted by the mobile application attached to the view it is generated in, or a crash of the application.                                                                                                                                                                                        |
```

| Attribute name     | Type   | Description                                                               |
|--------------------|--------|---------------------------------------------------------------------------|

> | `os.name`          | string | The OS name as reported by the by the device (System User-Agent).          |

Reviewer 3: ```suggestion
| `os.name`          | string | The OS name as reported by the device (System User-Agent).          |
```

> | `os.version`       | string | The OS version as reported by the by the device (System User-Agent).       |

Reviewer 3: ```suggestion
| `os.version`       | string | The OS version as reported by the device (System User-Agent).       |
```

> | `os.version_major` | string | The OS version major as reported by the by the device (System User-Agent). |

Reviewer 3: ```suggestion
| `os.version_major` | string | The OS version major as reported by the device (System User-Agent). |
```
